src/nvim/api/vim.c

```diff
@@ -737,9 +738,44 @@ Object nvim_get_option_value(String name, Dict(option) *opts, Error *err)
     goto end;
   }
 
+  int opt_type = SREQ_GLOBAL;
+  void *from = NULL;
+
+  if (opts->win.type == kObjectTypeInteger) {
```

Comment:
Again, the argument validation is quite similar to `nvim_set_option_value`. Can we unify?